.github/workflows/rails_tests.yml

```diff
@@ -117,36 +126,31 @@ jobs:
 
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
+
       - name: Set up dependencies
         run: |
-          apk add --no-cache \
-            nodejs yarn npm \
-            tzdata \
-            git \
-            bash \
-            freetds-dev \
-            icu icu-dev \
-            curl libcurl curl-dev \
-            imagemagick \
-            libmagic file-dev file \
-            build-base libxml2-dev libxslt-dev postgresql-dev \
-            libgcc libstdc++ libx11 glib libxrender libxext libintl ttf-dejavu ttf-droid ttf-freefont ttf-liberation ttf-freefont \
-            chromium nss freetype freetype-dev harfbuzz ca-certificates ttf-freefont \
-            lftp postgresql tmux postgis geos geos-dev \
-            shared-mime-info gpg gpg-agent
-
+          apk add --no-cache $(cat .github/dependencies.txt)
```

Comment:
This seems alright, minor concern about (1) making it harder to find the git history around why these things were added, and (2) no longer able to add comments alongside each dependency (not that we were doing that :) )

Reply:
These deps are repeated in several of the workflow files, my plan was to DRY up those files too but I could back this out if that's a concern. Also, having the deps in a file lets generate a checksum and cache the apks to avoid needing to download them on every run

Reply:
Given that each dependency is now on it's own line, could we not add a bash style comment if we wanted to for any line?  I'm unclear how apk add would handle that, but I'm sure we could put some awk/sed in place to strip them if we wanted the. I like that all of the dependencies will stay in sync, so at least that's a win, and they ARE still in git, so we could have commit comments if nothing else.

Reply:
Since we don't actually have any comments I was going to leave it as-is until it's a need. But we could use sed/awk or just make it a script itself (`sh ./github/install-deps.sh`)

Reply:
[tracking issue here](https://github.com/open-path/Green-River/issues/6409)